From a066306d783693d3f78b9c5e84feca7d690cf27a Mon Sep 17 00:00:00 2001
From: jiangheng <jiangheng12@huawei.com>
Date: Fri, 2 Jul 2021 16:54:43 +0800
Subject: [PATCH] fix coredump in etharp

---
 src/core/ipv4/etharp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/ipv4/etharp.c b/src/core/ipv4/etharp.c
index c3a5a10..effb7db 100644
--- a/src/core/ipv4/etharp.c
+++ b/src/core/ipv4/etharp.c
@@ -102,10 +102,10 @@ struct etharp_entry {
   u8_t state;
 };
 
-static struct etharp_entry arp_table[ARP_TABLE_SIZE];
+static PER_THREAD struct etharp_entry arp_table[ARP_TABLE_SIZE];
 
 #if !LWIP_NETIF_HWADDRHINT
-static netif_addr_idx_t etharp_cached_entry;
+static PER_THREAD netif_addr_idx_t etharp_cached_entry;
 #endif /* !LWIP_NETIF_HWADDRHINT */
 
 /** Try hard to create a new entry - we want the IP address to appear in
-- 
2.23.0

