From 1ede4a00c1eca575314af02374846cb086798c08 Mon Sep 17 00:00:00 2001
From: wu-changsheng <wuchangsheng2@huawei.com>
Date: Tue, 20 Dec 2022 14:37:21 +0800
Subject: [PATCH] add variable in struct sock

---
 src/include/lwipsock.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/include/lwipsock.h b/src/include/lwipsock.h
index ec4d78c..8924728 100644
--- a/src/include/lwipsock.h
+++ b/src/include/lwipsock.h
@@ -105,8 +105,7 @@ struct lwip_sock {
   char pad1 __rte_cache_aligned;
   /* app and stack thread all use */
   uint32_t in_send; /* avoid sock too much send rpc msg*/
-  pthread_spinlock_t sock_lock;
-
+  bool read_wait;
   char pad2 __rte_cache_aligned;
   /* stack thread all use */
   struct list_node recv_list;
-- 
2.23.0

