From 5527e02b7ae7f27db8964ad55747326b98e33634 Mon Sep 17 00:00:00 2001
From: wu-changsheng <wuchangsheng2@huawei.com>
Date: Wed, 21 Dec 2022 22:47:04 +0800
Subject: [PATCH] reduce struct pbuf size

---
 src/include/lwip/pbuf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/lwip/pbuf.h b/src/include/lwip/pbuf.h
index 380ac8e..1124408 100644
--- a/src/include/lwip/pbuf.h
+++ b/src/include/lwip/pbuf.h
@@ -229,7 +229,7 @@ struct pbuf {
   u64_t l3_len:9;
   /* < L4 (TCP/UDP) Header Length. */
   u64_t l4_len:8;
-  u16_t header_off;
+  u8_t header_off;
   u8_t rexmit;
   u8_t in_write;
   u8_t head;
-- 
2.23.0

