From 0e16f4ec71b0794f48cb7b9e99712c36e40d4d48 Mon Sep 17 00:00:00 2001
From: kircher <majun65@huawei.com>
Date: Wed, 22 Mar 2023 15:16:04 +0800
Subject: [PATCH] lwip-send-recv-thread-bind-numa

---
 src/include/lwipsock.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/include/lwipsock.h b/src/include/lwipsock.h
index f917d8a..f8480c5 100644
--- a/src/include/lwipsock.h
+++ b/src/include/lwipsock.h
@@ -138,6 +138,7 @@ struct lwip_sock {
   const struct rte_memzone *same_node_rx_ring_mz;
   struct same_node_ring *same_node_tx_ring;
   const struct rte_memzone *same_node_tx_ring_mz;
+  uint8_t already_bind_numa;
 #endif
 };
 
-- 
2.33.0

