From b94a7024bc7dc4984039b4f54aff3dbdcd21d8b8 Mon Sep 17 00:00:00 2001
From: jiangheng12 <jiangheng14@huawei.com>
Date: Wed, 14 Jun 2023 18:34:12 +0800
Subject: [PATCH] fix udp recvmbox size not set

---
 src/include/lwipopts.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/include/lwipopts.h b/src/include/lwipopts.h
index 6b5c769..f0df0e3 100644
--- a/src/include/lwipopts.h
+++ b/src/include/lwipopts.h
@@ -180,6 +180,7 @@
    ---------------------------------
 */
 #define LWIP_UDP 1
+#define DEFAULT_UDP_RECVMBOX_SIZE 4096
 
 
 /*
-- 
2.23.0

